var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -73,10 +74,34 @@ def configure_args(self):
 
         return args
 
-    @run_before('install')
+    @run_before('configure')
+    def fix_python_path(self):
+        if not self.spec.satisfies('@2.53.4:'):
+            return
+
+        files = ['gobject/glib-genmarshal.in', 'gobject/glib-mkenums.in']
+
+        filter_file('^#!/usr/bin/env @PYTHON@',
+                    '#!/usr/bin/env python',
+                    *files)
+
+    @run_after('install')
     def filter_sbang(self):
-        # Filter sbang before install so Spack's sbang hook can fix it up
-        perl = join_path(self.spec['perl'].prefix.bin, 'perl')
-        files = ['gobject/glib-mkenums']
+        # Revert sbang, so Spack's sbang hook can fix it up (we have to do
+        # this after install because otherwise the install target will try
+        # to rebuild files as filter_file updates the timestamps)
+        if self.spec.satisfies('@2.53.4:'):
+            python = join_path(self.spec['python'].prefix.bin, 'python')
```

Comment:
This only works for Python 2. Please use `self.spec['python'].command.path` instead.